Eastern-Research-Group/mywaterway
# Feature/add data widget

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3544126

## Main Changes:
* Adds the add data widget to HMW. This was modeled after the add data widget that is apart of Esri's web app builder and TOTS.  
  * See https://geopub.epa.gov/DWWidgetApp/# for an example of the add data widget from Esri's web app builder.
  * See the "Add Data" tab on https://app7.erg.com/totsdev/ for the TOTS version of the add data widget.

## Steps To Test:
1. Navigate to the community page
2. Open the add data widget
3. Test closing the add data widget by clicking the "X" button and by clicking the add data widget button
4. Add layers via the "Search" panel
  * Note: Some layers require you to login with an ArcGIS Online account. If you don't have one you can toggle the "Within map..." switch to get more layers that do not require login.
5. Add layers via the "URL" panel. There are sample urls available on the panel, if you don't have any URLS on hand. 
6. Add layers via the "File" panel. Feel free to contact me, if you need some files to test this. 
7. Verify the widget works from a mobile device. The widget should dock to the bottom on smaller screens.
8. Verify the widget works from the state page
9. Verify the widget works from the plan summary page
10. Verify the widget works from the waterbody report page
